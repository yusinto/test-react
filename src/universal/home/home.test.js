import React from 'react';
import {shallow} from 'enzyme';
import HomeRedux from './home';
import td from 'testdouble';

describe('Home component tests', () => {
  // use rewire to extract private class
  const Home = HomeRedux.__get__('Home');

  it('should render correctly', () => {
    const output = shallow(<Home randomNumber={45}/>);
    expect(output).toMatchSnapshot();
  });

  it('should invoke private generateRandom method on button click', () => {
    // create a mock function using testdouble and set it up
    // to return a predictable value when invoked
    const mockGenerateRandom = td.function('mockGenerateRandom');

    // replace the real function with our mock
    Home.prototype.onClickGenerateRandom = mockGenerateRandom;

    // render the component
    const output = shallow(<Home randomNumber={45}/>);
    const button = output.find('button');

    // simulate button click to invoke the mock function
    button.simulate('click');

    // assert the mock function was invoked
    td.verify(mockGenerateRandom());
    expect(output).toMatchSnapshot();
  });
});