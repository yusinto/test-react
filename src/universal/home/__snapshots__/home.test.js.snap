exports[`Home component tests should invoke private generateRandom method on button click 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <p>
        Click button below to generate a random number!
    </p>
    <div>
        45
    </div>
    <button
        onClick={[Function]}>
        Generate random number
    </button>
</div>,
  "nodes": Array [
    <div>
      <p>
            Click button below to generate a random number!
      </p>
      <div>
            45
      </div>
      <button
            onClick={[Function]}>
            Generate random number
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Home
        randomNumber={45} />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Home {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onClickGenerateRandom": [Function],
        "props": Object {
          "randomNumber": 45,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <p>
                    Click button below to generate a random number!
          </p>
          <div>
                    45
          </div>
          <button
                    onClick={[Function]}>
                    Generate random number
          </button>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <p>
                    Click button below to generate a random number!
          </p>
          <div>
                    45
          </div>
          <button
                    onClick={[Function]}>
                    Generate random number
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home
    randomNumber={45} />,
}
`;

exports[`Home component tests should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <p>
        Click button below to generate a random number!
    </p>
    <div>
        45
    </div>
    <button
        onClick={[Function]}>
        Generate random number
    </button>
</div>,
  "nodes": Array [
    <div>
      <p>
            Click button below to generate a random number!
      </p>
      <div>
            45
      </div>
      <button
            onClick={[Function]}>
            Generate random number
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Home
        randomNumber={45} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Home {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onClickGenerateRandom": [Function],
        "props": Object {
          "randomNumber": 45,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <p>
                    Click button below to generate a random number!
          </p>
          <div>
                    45
          </div>
          <button
                    onClick={[Function]}>
                    Generate random number
          </button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <p>
                    Click button below to generate a random number!
          </p>
          <div>
                    45
          </div>
          <button
                    onClick={[Function]}>
                    Generate random number
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home
    randomNumber={45} />,
}
`;
